### PR TITLE
[#872][FOLLOWUP] feat(tez): Add UmbilicalUtils to get Worker info from AM

### PR DESCRIPTION
**Pull request by lifeSo**
### What changes were proposed in this pull request?

Add UmbilicalUtils to get Worker info from AM
### Why are the changes needed?

Fix: https://github.com/apache/incubator-uniffle/issues/872

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

unit test

### CODE REVIEW

File: client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    GetShuffleServerResponse response = umbilical.getShuffleAssignments(request);
+    Map<Integer, List<ShuffleServerInfo>>  partitionToServers = response.getShuffleAssignmentsInfoWritable()
+        .getShuffleAssignmentsInfo()
+        .getPartitionToServers();
+    LOG.info("RequestShuffleServer applicationId:{}, taskAttemptId:{}, host:{}, port:{}, shuffleId:{}, worker:{}",
+        applicationId, taskAttemptId, amHostPort.getLeft(), amHostPort.getRight(), shuffleId, partitionToServers);
+    return partitionToServers;
+  }
+
+  public static Map<Integer, List<ShuffleServerInfo>> requestShuffleServer(
```
Comment (jerqi) [inline client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java:99]: Could you add UT for this class?

Comment (lifeSo) [reply client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java:99]: ok, it is not easy as early, we will add later.

Comment (jerqi) [reply client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java:99]: OK.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @lifeSo

File: client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  private static Map<Integer, List<ShuffleServerInfo>> doRequestShuffleServer(
+      ApplicationId applicationId,
+      Configuration conf,
+      TezTaskAttemptID taskAttemptId,
+      int shuffleId) throws IOException, InterruptedException, TezException {
+    UserGroupInformation taskOwner = UserGroupInformation.createRemoteUser(applicationId.toString());
+
+    Pair<String, Integer> amHostPort = getAmHostPort();
+    final InetSocketAddress address = NetUtils.createSocketAddrForHost(amHostPort.getLeft(), amHostPort.getRight());
```
Comment (jerqi) [inline client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java:78]: Could we extract the client for this method like
```
doRequestShuffleServer(
      ApplicationId applicationId,
      Configuration conf,
      TezTaskAttemptID taskAttemptId,
      int shuffleId,
     TezRemoteShuffleUmbilicalProtocol umbilical)
```
We could use mock client to add UT.

Comment (lifeSo) [reply client-tez/src/main/java/org/apache/tez/common/UmbilicalUtils.java:78]: ok, we format the code in :

https://github.com/apache/incubator-uniffle/pull/920/commits/3c374bbc248cd2d411235c205d5e804c8dba7002

and we may modify the methed to add unit test.

thanks for merge

**Pull request merged**